File: front/EGAR/OrderTrackingApplication/src/types/order.ts
export type OrderStatus = 'Created' | 'Sent' | 'Delivered' | 'Cancelled';

export interface Order {
  id: string;
  orderNumber: string;
  description: string;
  status: OrderStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderRequest {
  orderNumber: string;
  description: string;
}

export interface UpdateOrderStatusRequest {
  status: OrderStatus;
}

export interface OrderStatusNotification {
  orderId: string;
  orderNumber: string;
  previousStatus: OrderStatus;
  newStatus: OrderStatus;
  occurredAt: string;
}

File: c#/EGAR/OrderTrackingApplication/src/OrderTrackingApplication.Domain/Entities/Order.cs
using OrderTrackingApplication.Domain.Enums;

namespace OrderTrackingApplication.Domain.Entities;

public class Order : Entity
{
    public string OrderNumber { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public OrderStatus Status { get; set; } = OrderStatus.Created;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public void UpdateStatus(OrderStatus newStatus)
    {
        if (Status == newStatus)
        {
            return;
        }

        Status = newStatus;
        UpdatedAt = DateTime.UtcNow;
    }
}
